### PR TITLE
ENH: Synchronize gbq with upstream

### PR DESCRIPTION
**Pull request by bashtage**
Align with actual API

### CODE REVIEW

**Comment by bashtage** [comment]
No tests due to optional dependencies and extra CI overhead of trying to test against big query.

File: pandas-stubs/io/gbq.pyi [hunk 1]
```diff
@@ -12,10 +12,10 @@ def read_gbq(
     dialect: str | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/io/gbq.pyi:12]: `dialect` is `'standard'` or `'legacy'` according to the docs

File: pandas-stubs/io/gbq.pyi [hunk 2]
```diff
@@ -12,10 +12,10 @@ def read_gbq(
...
     location: str | None = ...,
     configuration: dict[str, Any] | None = ...,
-    credentials=...,
+    # Google type, not available
+    credentials: Any = ...,
     use_bqstorage_api: bool | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/io/gbq.pyi:17]: according to docs, `use_bqstorage_api` can't be `None`

File: pandas-stubs/io/gbq.pyi [hunk 3]
```diff
@@ -12,10 +12,10 @@ def read_gbq(
...
-    private_key=...,
-    verbose=...,
+    max_results: int | None = ...,
     progress_bar_type: str | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/io/gbq.pyi:19]: `progress_bar_type` can be `Literal["tqdm", "tqdm_notebook", "tqdm_gui"]` according to docs

File: pandas-stubs/io/gbq.pyi [hunk 4]
```diff
@@ -12,10 +12,10 @@ def read_gbq(
...
 ) -> DataFrame: ...
 def to_gbq(
```
Comment (Dr-Irv) [inline pandas-stubs/io/gbq.pyi:21]: `pandas.io.gbq` isn't public, so this should just be in `frame.pyi`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

**Comment by Dr-Irv** [comment]
> No tests due to optional dependencies and extra CI overhead of trying to test against big query.

Here I agree with you - the overhead of testing against big query isn't worth it.

File: pandas-stubs/io/gbq.pyi (hunk 2 above)
Comment (bashtage) [reply pandas-stubs/io/gbq.pyi:17]: The default value is `None`, so it clearly is accepted.

Comment (Dr-Irv) [reply pandas-stubs/io/gbq.pyi:20]: Probably worthwhile to create an issue in the `pandas` repo

**Comment by Dr-Irv** [comment]
@bashtage can you avoid doing force pushes, as it makes it hard to see the previous comments I made within the code changes you made?

**Review by Dr-Irv — APPROVED** [review]
Thanks @bashtage

**Comment by bashtage** [comment]
I only did a force because I rebased.  Commits are linear

**Comment by Dr-Irv** [comment]
> I only did a force because I rebased. Commits are linear

I don't think you have to do a force when you rebase.  Whereas the commits are linear, github loses the connection between my previous comments and the forced push version.

**Pull request merged**